src/matter/interaction/InteractionServer.ts

```diff
@@ -252,21 +323,33 @@ export class InteractionServer implements ProtocolHandler<MatterDevice> {
         return `${endpointName}/${clusterName}/${attributeName}`;
     }
 
+    private getAttribute(path: Partial<Path> ): AttributeWithPath[] {
```

Comment:
getAttribute and getAttributes methods are confusing because both returns a list of attributes so it unclear was each one does.

I think the intent is this one:
* one will ignore unknown attributes (for read)
* the other one will report an error for unknown attributes (for write)

Maybe it would be better to keep a single method and a an extra argument to indicate which behavior is expected?
undefined can be return for unknown attribute so it can be converted into an error instead of ignoring it.
It is unclear what should the behavior for "star" IDs (undefined) which defines a filter. If no attributes are defined for this filter, should we return an error for the filter or ignore it? Is it even allow to write to a "star" pattern? (Like */34/2)

Reply:
ignore , I adjust it back :-)